### ISSUE TITLE
Android Client - unable to star contacts

### ISSUE THREAD

**Issue by zstephen** [bitfireAT/davdroid#408]
Running Software:
Samsung Galaxy S2 GT-i9100 with Slim-4.4.2.build-5-OFFICIAL-4068
DAVdroid 0.6.8 installed from the F-droid app
Ubuntu 14.04.1 LTS running Zimbra 8.6 FOSS edition

On my mobile I can sync contacts, but I am unable to star any contacts as like can be done with ActiveSync and local contacts.

When I click star is momentarily shows the star shaded, then returns back to the outlined star.

I'm not sure if this is a limitation of a field existing on the Zimbra server that the DAVdroid needs to utilise or an issue with DAVdroid itself, so please let me know and I'll produce the error with the relevant logs on my device.

Thanks

**Comment by Trim** [comment]
I'm having same issue with my Owncloud account.

I've noticed that if I create a new contact from DAVdroid or Gnome Contacts or Evolution and star it, it works.

Although, if I'm starring an old contact it doesn't work (same behaviour than bug report).

**Comment by daks** [comment]
FYI: I just tried with last version of DAVDroid connected to a Radicale (http://radicale.org) server, and it works.

**Comment by vacy** [comment]
Same issue with owncloud 7.0.4 on Phone OnePlus One - Cyanogenmod 11 M11, Android 4.4.4 here.
I guess it is not a problem with the carddav backend because the sync doesn't start just in that moment, but later on.
an "adb logcat" on this action did not post any output.

**Comment by rfc2822** [comment]
Please provide steps to reproduce. Also note that "starring" is not defined in VCard, so DAVdroid uses the custom property [X-DAVDROID-STARRED](https://github.com/bitfireAT/davdroid/blob/6c998c31c358b1330c19764d6f3620cc3a832abd/app/src/main/java/at/bitfire/davdroid/resource/Contact.java#L69) which if course can't be understood by other clients. Other custom propertys by other clients (maybe Evolution) are preserved, but ignored.

**Comment by philenotfound** [comment]
I have a similar problem with davdroid + owncloud, I can't star old exisiting contacts, if i create a new one with the davdroid account it works. If I click the star on an old contact it turns white, like it's supposed to, but I don't see anything in the logcat with regard to that action. But if I add "X-DAVDROID-STARRED:1" to an old contact manually in the MySQL database entry for the old contact it gets synced to the device once I star a new contact and initiate a sync from the device. 
So I end up with a new and an old contact starred..I can't unstar the old contact again, if I try that the star turns empty but upon entering the contact again, the star is solid again.
Staring/unstaring the new contact works fine though.

**Comment by 4lexwe** [comment]
Sorry, to bring this up again! @rfc2822

**Problem:** Can't 'star' contacts

**Steps to reproduce:**
* Open contacts app
* Select contact, only associated with davdroid **(no Whatsapp, Threema, etc.)**
* 'Star' the contact
* Star icon turns white
* Go back and look at the 'starred contacts'

**Expectation:** Contact is in the 'starred contacts' and if you open the contact, the star icon is white.

**Reality:** Contact isn't starred and if you open the contact, the star icon is not white.

**Additional information:**
* *Device:* OnePlus One
* *OS:* CyanogenMod 12.1 (Nightly-build 18.04.2015)
* *Used App:* com.android.contacts
* *CalDav-Server:* ownCloud 8.0
* If the contact is associated with third party apps (like Whatsapp, Threema, etc.) the expectation is achieved.

**Comment by Cameleon** [comment]
Hi,
Just wanted to say that "I got this problem too" with an owncloud 8 server provided by zaclys (french provider). Hope something can be done.

**Comment by untitaker** [comment]
Could it be that another client drops the unknown property?

**Comment by irgendwie** [comment]
@untitaker I don't think so, as it works for contacts, which are connected to other apps - but not with contacts, only used by DavDroid.

**Comment by androdid** [comment]
Hello,

I had the same problem on an Xperia Z3C and CM12.1, using DAVDROID to synchronize my contacts with my ownCloud server. And before with my S3mini and the same OS.
I was unable to star or unstar my contacts with the contact defaut app, but it was working with Contact+

I think i found the problem, it's come from the IUD of the contacts.
When UID is of the form XXXXXXXXXXXX.XXXXXXXXXXX, it works.
When UID is of the form XXXXXXXXXXXX.XXXXXXXXXXX.vcf
or XXXXXXXXXXXX.XXXXXXXXXXX@myowncloudserver.com
it doesn't.
The problem is the dot and the extension at the end of your contacts.

So, to solve the probem, I exported all my contacts on a vcf file. Edited it with a text editor to delete all "@myowncloudserver.com" and ".vcf" (use the find and replace fonction), and then imported my contacts again on my server, 
I resynchronized my phone and everthing is working fine.

**Comment by untitaker** [comment]
@androdid Do you experience the same issues again if you re-add those parts to the UID? If you have some items with the `@`-parts in the UID, and some not, do these exhibit different behavior when starred?

**Comment by androdid** [comment]
Yes I've tried to re-add the @ part on some contacts, and I have the same issue.
It looks like when a contact has a @ part (or even just .vcf), contact app can't edit it and add or remove the X-DAVDROID-STARRED:1 line

Contact+ doesn't have this problem.
